crates/services/p2p/src/discovery.rs

```diff
@@ -265,7 +268,10 @@ mod tests {
                 .with_bootstrap_nodes(bootstrap_nodes)
                 .with_random_walk(Duration::from_millis(500));
 
-            config.finish()
+            async move { config.finish().await }
+                .now_or_never()
+                .unwrap()
+                .unwrap()
```

Comment:
I don't see how we can guarantee this future to be immediately ready. What happens if the `DnsResolver::new().await?` is pending for example? As I read this, we'd panic in that scenario. That feels quite brittle to me, in which case it would feel safer to just use the synchronous resolver instead.

My preferred option though would be to allow `build_behavior_fn` to return a function which returns a future, so we can return this future without having to panic if it isn't ready here.

Reply:
deprecated in 5b4671f1f

Reply:
Nice, thank you!